arangod/Aql/ExecutionEngine.cpp

```diff
@@ -648,6 +648,13 @@ Result ExecutionEngine::shutdownSync(int errorCode) noexcept try {
         if (state == ExecutionState::WAITING) {
           sharedState->waitForAsyncWakeup();
         }
+        if ((errorCode == TRI_ERROR_INTERNAL || errorCode == TRI_ERROR_SHUTTING_DOWN) && 
```

Comment:
From the comment below, I would expect this be added 3 lines above before the test for WAITING.

IF there is no one to wake us up, there is no point in waiting for someone.
If may be excluded by the behavior of waitForAsyncWakeup, but I would like to not depend on it here.

Reply:
this seems unelegant, maybe throw in waitForAsyncWakeup if the shared state is not valid ?